plugin-examples/src/test/java/com/hotels/styx/ExampleMessageInterceptTest.java

```diff
@@ -0,0 +1,34 @@
+package com.hotels.styx;
+
+import com.hotels.styx.api.*;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+/**
+ * This tests the behaviours added in the example plugin.
+ */
+
+public class PluginExampleTest {
```

Comment:
Java class names must match the file name. In the case of tests, we take the name of the class under test and add `Test` to the end.